lib/httpserver/httpserver.go

```diff
@@ -430,7 +431,7 @@ func CheckAuthFlag(w http.ResponseWriter, r *http.Request, flagValue string, fla
 	if flagValue == "" {
 		return CheckBasicAuth(w, r)
 	}
-	if r.FormValue("authKey") != flagValue {
+	if subtle.ConstantTimeCompare([]byte(r.FormValue("authKey")), []byte(flagValue)) != 1 {
```

Comment:
Based on `ConstantTimeCompare` description:
```
// ConstantTimeCompare returns 1 if the two slices, x and y, have equal contents
// and 0 otherwise. The time taken is a function of the length of the slices and
// is independent of the contents. If the lengths of x and y do not match it
```

It is not enough to use `subtle.ConstantTimeCompare` solely, it is also required to use `ConstantTimeEq` to compare slices len before comparing an actual value. Could you move both checks into separate function of in this package like:
```
func ConsantTimeEqual(a, b []byte) bool {

// compare slices len

// compare slices value

}
```

Reply:
@zekker6 , thanks for reviewing this change!

Maybe I misunderstood the comment, but from the documentation, it looks like the `subtle.ConstantTimeCompare` does the initial length check. If the lengths of the provided slices are not the same, it immediately returns 0.

Let me know if I am missing something here...

https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/crypto/subtle/constant_time.go;drc=07b0c57a75d4b8c94b16983c5a78026ba46377ac;l=14

Reply:
You are right. The problem here is that since it immediately returns `0` an attacker can recover a key length by using timing information. In order to avoid this it is required to perform an additional check of slices length by using `subtle.ConstantTimeEq` before using `subtle.ConstantTimeCompare` to compare slices itself.